x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_notify_when.tsx

```diff
@@ -0,0 +1,238 @@
+      {
+        defaultMessage: 'On check intervals',
+      }
+    ),
+    'data-test-subj': 'onActiveAlert',
+    dropdownDisplay: (
+      <>
+        <strong>
+          <FormattedMessage
+            defaultMessage="On check intervals"
+            id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onActiveAlert.label"
+          />
+        </strong>
+        <EuiText size="s" color="subdued">
+          <p>
+            <FormattedMessage
+              defaultMessage="Actions run if rule conditions are met."
+              id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onActiveAlert.description"
+            />
+          </p>
+        </EuiText>
+      </>
+    ),
+  },
+  {
+    value: 'onThrottleInterval',
+    inputDisplay: i18n.translate(
+      'xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onThrottleInterval.display',
+      {
+        defaultMessage: 'On custom action intervals',
+      }
+    ),
+    'data-test-subj': 'onThrottleInterval',
+    dropdownDisplay: (
+      <>
+        <strong>
+          <FormattedMessage
+            defaultMessage="On custom action intervals"
+            id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onThrottleInterval.label"
+          />
+        </strong>
+        <EuiText size="s" color="subdued">
+          <p>
+            <FormattedMessage
+              defaultMessage="Actions run if rule conditions are met."
+              id="xpack.triggersActionsUI.sections.ruleForm.ruleNotifyWhen.onThrottleInterval.description"
+            />
+          </p>
+        </EuiText>
+      </>
+    ),
+  },
+];
+
+interface RuleNotifyWhenProps {
+  frequency: RuleAction['frequency'];
+  throttle: number | null;
+  throttleUnit: string;
+  onNotifyWhenChange: (notifyWhen: RuleNotifyWhenType) => void;
+  onThrottleChange: (throttle: number | null, throttleUnit: string) => void;
+}
+
+export const ActionNotifyWhen = ({
+  frequency = DEFAULT_FREQUENCY,
+  throttle,
+  throttleUnit,
+  onNotifyWhenChange,
+  onThrottleChange,
+}: RuleNotifyWhenProps) => {
+  const [ruleThrottle, setRuleThrottle] = useState<number>(throttle || 1);
+  const [showCustomThrottleOpts, setShowCustomThrottleOpts] = useState<boolean>(false);
+  const [notifyWhenValue, setNotifyWhenValue] =
+    useState<RuleNotifyWhenType>(DEFAULT_NOTIFY_WHEN_VALUE);
+
+  useEffect(() => {
+    if (frequency.notifyWhen) {
+      setNotifyWhenValue(frequency.notifyWhen);
+    } else {
+      // If 'notifyWhen' is not set, derive value from existence of throttle value
+      setNotifyWhenValue(frequency.throttle ? 'onThrottleInterval' : 'onActiveAlert');
+    }
+  }, [frequency]);
+
+  useEffect(() => {
+    setShowCustomThrottleOpts(notifyWhenValue === 'onThrottleInterval');
+  }, [notifyWhenValue]);
+
+  const onNotifyWhenValueChange = useCallback(
+    (newValue: RuleNotifyWhenType) => {
+      onNotifyWhenChange(newValue);
+      setNotifyWhenValue(newValue);
+      setTimeout(
```

Comment:
why do we need the setTimeout here?

Reply:
Needed to call both `onNotifyWhenChange` and `onThrottleChange` in the same function, otherwise React gets weird. Trying to avoid refactoring the entire form in order to make this PR work. I can add a comment

Reply:
make sense!, let's add a comment/issue to refactor it later on